auth/src/main/kotlin/com/hedvig/android/auth/AuthTokenServiceImpl.kt

```diff
@@ -0,0 +1,72 @@
+package com.hedvig.android.auth
+
+import com.hedvig.android.auth.storage.AuthTokenStorage
+import com.hedvig.authlib.AccessToken
+import com.hedvig.authlib.AuthRepository
+import com.hedvig.authlib.AuthTokenResult
+import com.hedvig.authlib.RefreshToken
+import com.hedvig.authlib.RefreshTokenGrant
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.mapLatest
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+class AuthTokenServiceImpl(
+  private val authTokenStorage: AuthTokenStorage,
+  private val authRepository: AuthRepository,
+  private val coroutineScope: CoroutineScope,
+) : AuthTokenService {
+
+  @Suppress("NAME_SHADOWING")
+  private val authStatus: StateFlow<AuthStatus?> = authTokenStorage.getTokens()
```

Comment:
Can this be public, and then we can remove `fun authStatus()`?

Reply:
Hmm yeah, why not actually, I made the change, I don't see a reason why this wouldn't be a good idea. Removes one useless function from this class!